src/api/app/controllers/webui/patchinfo_controller.rb

```diff
@@ -193,14 +193,43 @@ def save
   end
 
   def remove
+    # checks
+    if @package.name == '_project'
+      raise DeletePackageNoPermission.new '_project package can not be deleted.'
+    end
+
+    unless User.current.can_modify_package?(@package)
```

Comment:
That's a good idea. Though I would prefer to do that in a separate pull request.